numpy/core/arrayprint.py

```diff
@@ -305,24 +298,22 @@ def indirect(x):
 
     return formatdict
 
-def _get_format_function(data, precision, suppress_small, formatter):
+def _get_format_function(data, **options):
     """
     find the right formatting function for the dtype_
     """
     dtype_ = data.dtype
     if dtype_.fields is not None:
         format_functions = []
         for field_name in dtype_.names:
-            field_values = data[field_name]
-            format_function = _get_format_function(
-                    ravel(field_values), precision, suppress_small, formatter)
```

Comment:
This change affects formatting void scalars containing floats, right? (in a good way)

Reply:
I think `np.array((1.0,), np.dtype([('a', float)]))` might be the test, although I'm not sure what that did on your previous commit. On 1.13, that returns `( 1.0,)` with the space

Reply:
I don't quite follow.. in these particular lines (316-318) the only functional change is to remove a call to `ravel`. 

In 1.13 it is true that void scalars would never get here, however in 1.13 these 3 lines were copied in the `()` special case in `array2string`, so 1.13 ended up doing the same thing.

It's also true that spaces within void scalars and void arrays are gone now. I will add a test for that.